experimental/bridge/opentracing/lib.go

```diff
@@ -0,0 +1,558 @@
+	"fmt"
+	"net/http"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+
+	"google.golang.org/grpc/codes"
+
+	ot "github.com/opentracing/opentracing-go"
+	otext "github.com/opentracing/opentracing-go/ext"
+	otlog "github.com/opentracing/opentracing-go/log"
+
+	otelcore "go.opentelemetry.io/api/core"
+	otelregistry "go.opentelemetry.io/api/registry"
+	oteltrace "go.opentelemetry.io/api/trace"
+
+	migration "go.opentelemetry.io/experimental/bridge/opentracing/migration"
+)
+
+type bridgeSpanContext struct {
+	// TODO: have a look at the java implementation of the shim to
+	// see what do they do with the baggage items
+	baggageItems    map[string]string
+	otelSpanContext otelcore.SpanContext
+}
+
+var _ ot.SpanContext = &bridgeSpanContext{}
+
+func (c *bridgeSpanContext) ForeachBaggageItem(handler func(k, v string) bool) {
+	for k, v := range c.baggageItems {
+		if !handler(k, v) {
+			break
+		}
+	}
+}
+
+func (c *bridgeSpanContext) setBaggageItem(restrictedKey, value string) {
+	if c.baggageItems == nil {
+		c.baggageItems = make(map[string]string)
+	}
+	crk := http.CanonicalHeaderKey(restrictedKey)
+	c.baggageItems[crk] = value
+}
+
+func (c *bridgeSpanContext) baggageItem(restrictedKey string) string {
+	crk := http.CanonicalHeaderKey(restrictedKey)
+	return c.baggageItems[crk]
+}
+
+type bridgeSpan struct {
+	otelSpan oteltrace.Span
+	ctx      *bridgeSpanContext
+	tracer   *BridgeTracer
+}
+
+var _ ot.Span = &bridgeSpan{}
+
+func (s *bridgeSpan) Finish() {
+	s.otelSpan.Finish()
+}
+
+func (s *bridgeSpan) FinishWithOptions(opts ot.FinishOptions) {
+	var otelOpts []oteltrace.FinishOption
+
+	if !opts.FinishTime.IsZero() {
+		otelOpts = append(otelOpts, oteltrace.WithFinishTime(opts.FinishTime))
+	}
+	for _, record := range opts.LogRecords {
+		s.logRecord(record)
+	}
+	for _, data := range opts.BulkLogData {
+		s.logRecord(data.ToLogRecord())
+	}
+	s.otelSpan.Finish(otelOpts...)
+}
+
+func (s *bridgeSpan) logRecord(record ot.LogRecord) {
+	// TODO: The context shouldn't be stored anywhere, so where do
+	// we get it from? I think we will need to defer the creation
+	// of the span, or something…
+	s.otelSpan.AddEventWithTimestamp(context.TODO(), record.Timestamp, "", otLogFieldsToOtelCoreKeyValues(record.Fields)...)
+}
+
+func (s *bridgeSpan) Context() ot.SpanContext {
+	return s.ctx
+}
+
+func (s *bridgeSpan) SetOperationName(operationName string) ot.Span {
+	s.otelSpan.SetName(operationName)
+	return s
+}
+
+func (s *bridgeSpan) SetTag(key string, value interface{}) ot.Span {
+	switch key {
+	case string(otext.SpanKind):
+		// TODO: Should we ignore it?
+	case string(otext.Error):
+		if b, ok := value.(bool); ok {
+			status := codes.OK
+			if b {
+				status = codes.Unknown
+			}
+			s.otelSpan.SetStatus(status)
+		}
+	default:
+		s.otelSpan.SetAttribute(otTagToOtelCoreKeyValue(key, value))
+	}
+	return s
+}
+
+func (s *bridgeSpan) LogFields(fields ...otlog.Field) {
+	// TODO: The context shouldn't be stored anywhere, so where do
```

Comment:
Since you have the `otelSpan` reference, I think it's fine to pass `context.Background()` below. OpenTracing didn't support it, but the OTel semantics don't prescribe anything about context tags for span events. (Why is it used in OTel-Go, then?
(a) Maybe it should not be
(b) Maybe context is used for some other reason in an SDK)

Reply:
I'll file an issue about clarifying the purpose of the context. The result would either be some docs or maybe the removal of the parameter.